README.md

```diff
@@ -0,0 +1,308 @@
+
+
+## Organizing stylesheets
+
+
+A typical Angular component directory (a search bar directive, in this case) will look like this:
+
+```
+buildium-base/components/search
+
+-search.html           //template
+-search.js             //directive
+-search.less           //styles
+-search.test.js        //unit test
+
+```
+
+As you can see, this component contains it's own stylesheet.  This is imported into a file in the root components directory called main.less.
+
+components/main.less is then imported into a styles directory within buildium-base.  This style directory contains common elements like mixins, variables, and other tools that we'd like access to when writing component specific styles.  This approach keeps the component specific stylesheets lean, and also reserves a special place to write styles that can be used globally.
+
+##Common Variables
+
+Use common mixins for visual behaviors that are likely repeated throughout the site such as colors, box-shadow, border-radius, etc...
+
+All colors should be defined as common variables, and we store them in buildium-base/styles/variables/colors.less
+
+##Naming
+
+We use the Block Element Modifier (BEM) naming convention.  BEM-style names are, in general, very descriptive. They also provide a way of indicating hierarchy without the side-effects of nesting classes and causing specificity battles. Finally, BEM is very conducive to creating a component-based UI (which we strive for in our Angular-based code).
+
+block__element-modifier
+
+Always try to make class names meaningful -- "large" is better than 600, "primary" is better than blue.
+
+```css
+//less
+
+.search__entity-list {
+   .list-reset;
+}
+
+.search__entity-list-item {
+   padding: 10px 5px 10px 38px;
+   background-image: url(/Manager/static-images/search-sprite.png);
+   background-repeat: no-repeat;
+   font-size: 12px;
+}
+
+.search__entity-list-item--active {
+   background-color: @focusHighlight;
+}
+
+//view
+
+<ul class="search__entity-list">
+  <li class="search__entity-list-item  
+    search__entity-list-item--{{item.category}}"
+    ng-repeat="item in entity.item"
+    ng-class="{ 'search__entity-list-item--active': item === search.activeItem }"
+    ng-mouseover="search.activeItem = item">
+</ul>
+```
+
+It’s better to separate out what is essential to all form columns (positioning and gutters) and what is a specific modifier (width).
+
+```css
+//Avoid - does too many things
+
+.formColumnSmall {
+    float: left;
+    margin: 0 @grid-gutter-width / 2;
+    width: 90px;
+}
+
+//Instead, seperate it out...
+
+.form__column {
+   float: left;
+   margin: 0 @grid-gutter-width / 2;
+}
+
+.form__column--small {
+   width: 90px;
+}
+
+.form__column--default {
+   width: 202.5px;
+}
+```
+
+## Positioning/Layout
+
+Seperate positioning and layout from reuseable components.  We use a grid system for positioning, making things predictable and preventing a specific component's styles from affecting others.
```

Comment:
Should probably also have a comma after predictable (looks like we're following that style in other parts of the doc).